app/models/league.rb

```diff
@@ -38,13 +38,31 @@ class League < Event
   enum game_mode: [:round_robin, :two_halfs, :swiss, :danish]
 
   def generate_schedule
-    if game_mode == League.game_modes.key(0)
+    if game_mode == League.game_modes[:round_robin]
       calculate_round_robin
-    elsif game_mode == League.game_modes.key(1)
+    elsif game_mode == League.game_modes[:two_halfs]
       calculate_two_halfs
+    elsif game_mode == League.game_modes[:swiss]
```

Comment:
you should be able to access enum state via autogenerated accessors. `self.swiss?` should return true if the gamemode of this league is swiss. Try it out, more info here: https://hackhands.com/ruby-on-enums-queries-and-rails-4-1/

Reply:
Wow nice :) I'll do that.